lib/localhost/authority.rb

```diff
@@ -196,6 +197,7 @@ def load(path = @root)
 		end
 		
 		def save(path = @root)
+			migrate_legacy_authority_path(path)
 			Dir.mkdir(path, 0700) unless File.directory?(path)
```

Comment:
This line fails on external tests, where the XDG state directory doesn't exist.

I wonder if we should use `FileUtils.mkdir_p` here instead?

Should we be creating `.local` if it doesn't exist?

Reply:
See https://github.com/socketry/localhost/actions/runs/8656328284/job/23736689683?pr=28#step:4:248 for the actual failure.

Reply:
The question is, for users who don't use `.local`, is it appropriate for random applications to create it? Does XDG have guidelines we can follow?

Reply:
Yeah, I was thinking about that. It seems like we should. The basedir spec doesn't seem to have a strong opinion.

Reply:
Okay, what I'd suggest we do, is move that into a separate operation, at least initially.

Reply:
I think initially, let's detect if we need to do it in `Authority.path`.

Reply:
>  If, when attempting to write a file, the destination directory is non-existent an attempt should be made to create it with permission 0700. If the destination directory exists already the permissions should not be changed. The application should be prepared to handle the case where the file could not be written, either because the directory was non-existent and could not be created, or for any other reason. In such case it may choose to present an error message to the user. 

is what the spec says, immediately after discussing the *config* directory, but I think it applies to stateful files like this as well.

For non-Linux systems, like MacOS, XDG doesn't make sense and yet it also makes sense. (It seems like CLI tools will use XDG anyways, but graphical ones will use ~/Library/)